thunder/core/transforms.py

```diff
@@ -1761,6 +1761,31 @@ def var_backward(a, dim, correction, v, g):
     return (2 * g * (a - mean)) / normalization_scalar
 
 
+@register_augmented_forward(prims.PrimIDs.STD)
+def std_aug_fwd(a, dim, *, correction):
+    v = prims.std(a, dim, correction=correction)
+    return VJPDual((v,), (a, dim, correction, v))
+
+
+@register_backward(prims.PrimIDs.STD)
+def std_backward(a, dim, correction, s, g):
+    n_elem_reduced = a.numel() // s.numel() if a.numel() != 0 else 1
+    normalization_scalar = n_elem_reduced - correction
+    g = restore_reduced_dims(g, dim, a.shape)
+    if a.dtype != s.dtype:
+        a = prims.convert_element_type(a, s.dtype)
+    mean = prims.sum(a, dim) / n_elem_reduced
+    mean = restore_reduced_dims(mean, dim, a.shape)
+
+    if len(dim) == 1:
+        dim = dim[0]
+    if dim != 0:
+        s = s.unsqueeze(dim)
```

Comment:
Looks like the tests are passing now.  Great!  But I do wonder: what's the difference between this and `s = restore_reduced_dims(s, dim, a.shape)`?

Reply:
my bad.

fixed!